test/embind/embind.test.js

```diff
@@ -449,33 +449,6 @@ module({
             assert.equal('ABCD', e);
         });
 
-        test("can pass Uint8Array to std::basic_string<unsigned char>", function() {
-            var e = cm.emval_test_take_and_return_std_basic_string_unsigned_char(new Uint8Array([65, 66, 67, 68]));
-            assert.equal('ABCD', e);
-        });
```

Comment:
Should we change the test such that it tests that this does not work?

Reply:
I tried that, and to do that I needed to revive `emval_test_take_and_return_std_basic_string_unsigned_char` in `embind_test.cpp` and the error message is like `Cannot call emval_test_take_and_return_std_basic_string_unsigned_char due to unbound types: NSt3__212basic_stringIhNS_11char_traitsIhEENS_9allocatorIhEEEE`, so the first test would be like
```
        test("cannot pass Uint8Array to std::basic_string<unsigned char>", function() {       
            var e = assert.throws(TypeError, function() {                        
              cm.emval_test_take_and_return_std_basic_string_unsigned_char(new Uint8Array([65, 66, 67, 68]));      
            });                                                                  
            assert.equal('Cannot call emval_test_take_and_return_std_basic_string_unsigned_char due to unbound types: NSt3__212basic_stringIhNS_11char_traitsIhEENS_9allocatorIhEEEE', e.message);
        }); 
```

Are you suggesting to change five deleted tests here like this?

Reply:
Ping 😀 By the way I wasn't able to get that code to work, it says the throwing error kind is different...

Reply:
Ping ping 😀 I think we can just delete this. The libc++ change means they don't allow invalid specializations anymore and `unsigned char` is one of many possible invalid specializations after all (Even `basic_string<int>` was possible before), and I don't think we can cover all invalid specializations by error tests anyway.